pool.go

```diff
@@ -55,7 +54,7 @@ type bufp struct {
 //
 // If no suitable buffer exists in the pool, Get creates one.
 func (p *BufferPool) Get(length int) []byte {
-	if length == 0 {
+	if length <= 0 {
```

Comment:
This should produce a big fat warning (that includes a stack trace). Can you add that (using the `go-log@v2` package that we use across other libp2p repos)?

Reply:
Thinking about this, we should probably `panic` here. Calling `make([]byte, l)` with a negative `l` also panics (https://go.dev/play/p/LFUw-YPlOq1), and this is just a fancy wrapper around that function.
Or even better, we could jump into the `if` below and have that one panic, as this will produce the more relevant error message.

Reply:
Unfortunately that assumption of a panic isn't correct because the only time that make receives the argument as is is when length > 0; in the case of a negative value, it gets converted into uint32 and a wrapped around large value which is the cause of the issue I reported.

Reply:
I meant this PR should change the code such that we panic.

Reply:
> the only time that make receives the argument as is is when length > 0; in the case of a negative value, it gets converted into uint32 and a wrapped around large value which is the cause of the issue I reported

`BufferPool` is an exported type, and `Get` an exported method. Consequently, nothing prevents `BufferPool.Get` from receiving a negative `length`.

Reply:
thats why you should panic; it is a case of a bad programming bug and it needs to be fixed.

Reply:
Alright, I've added a panic, please take a look and thanks y'all!

Reply:
Can you implement it as I suggested in https://github.com/libp2p/go-buffer-pool/pull/27/files#r886097258? It’s better to surface the proper Go panic to the user.

Reply:
@marten-seemann I did change it to invoke a panic with the length indicated per
<img width="598" alt="Screen Shot 2022-06-26 at 7 25 09 PM" src="https://user-images.githubusercontent.com/4898263/175828731-3de31106-da18-471f-b242-97ec0ba010fb.png">
I believe that suffices as far as negative lengths and indicating to the user what happened goes.

Are you asking that I pass in a negative length to make? If so, I am not comfortable making that change as that to me is a security issue to pass negative values into make, and if you'd like that change, I can close this PR and you can make that one :-)

Reply:
Why would that be a security issue? It will just panic.

Reply:
There are cases I may or may not know in which that can turn into a big allocation. Anyways I filed the issue, I've provided a PR, please feel free to choose the next course of action :-)

Reply:
It is guaranteed to panic: https://github.com/golang/go/blob/416c953960a475b7418b5c6aef0f46dd102b9129/doc/go_spec.html#L7187

Reply:
All yours to make the change, I can close my PR :-)